### PR TITLE
Implement symmetry in simulation/config

### PR DESCRIPTION
**Pull request by bruxillensis**
This PR adds a simulation symmetry in the simulation config, as a 3-tuple for x, y, and z. Options for the tuple are +1, 0, -1 and splits a full domain into a half/quarter/eighth and applies PEC/PMC at the symmetry boundary. Symmetry is passed to detectors/sources (including tidy3d mode solve). The user can build a full domain as normal and symmetry is applied automatically. The resulting fields are still the reduced domain, but can be unfolded with helpers unfold_fields / unfold_array / unfold_detector_states / unfold_source_mode.

Additional part of this PR - new rectilinear grid had a bug in is_uniform with atol=1=1e-8 (10nm), which given fdtd domains can sometimes have fine resolution it be incorrect. Changed to relative tolerance to calculate is_uniform.

## Summary by CodeRabbit

* **New Features**
  * Mirror-symmetry simulation support that reduces domain size (up to 8×) with automatic PEC/PMC-like symmetry walls and required unfolding workflow
  * Symmetry-aware mode solver, sources, and detectors to match reduced-domain boundary conditions
* **Documentation**
  * Detailed guide for simulation symmetry, requirements, usage, pitfalls, and unfolding/post-processing steps
* **Tests**
  * New unit and integration tests validating symmetry reduction, unfolding, grid behavior, and error cases

### CODE REVIEW

File: src/fdtdx/fdtd/symmetry.py [hunk 1]
```diff
@@ -0,0 +1,703 @@
...
+
+    # Per-axis absolute index of the symmetry plane (= shift applied to map full → reduced),
+    # and the reduced volume slice.
+    mid_abs = [0, 0, 0]
+    new_vol: list[tuple[int, int]] = []
+    for a in range(3):
+        vs0, vs1 = vol_slice[a]
+        n = vs1 - vs0
+        if symmetry[a] != 0:
+            if n < 2 or n % 2 != 0:
```
Comment (ymahlau) [inline src/fdtdx/fdtd/symmetry.py]: It might be worthwhile to refactor this error into a seperate function and use it across files

File: src/fdtdx/objects/detectors/mode.py [hunk 2]
```diff
@@ -50,6 +50,14 @@ class ModeOverlapDetector(PhasorDetector):
     #: curvature. Must differ from the propagation axis. Required when bend_radius is set.
     bend_axis: int | None = frozen_field(default=None)
 
+    #: Symmetry-plane condition at the min edge of each transverse axis (the two non-propagation
+    #: physical axes, in increasing-index order): ``0`` = PEC mirror (electric wall, the default),
+    #: ``1`` = PMC mirror (magnetic wall). Set this when the detector plane's waveguide lies on a
+    #: symmetry plane of a reduced (half/quarter) domain so the reference mode is solved with the
+    #: same wall the FDTD uses (e.g. ``(0, 1)`` for PEC at y=0 and PMC at the z Si-mid plane).
+    #: Must match the corresponding ModePlaneSource for a consistent overlap.
+    symmetry: tuple[int, int] = frozen_field(default=(0, 0))
```
Comment (ymahlau) [inline src/fdtdx/objects/detectors/mode.py:59]: I think a better way to do this would be to detect during placement if the mode detector/source touches the symmetry axis. This would avoid the extra effort of a user to always specify the symmetry for mode sources/detectors again.

**Comment by ymahlau** [comment]
@coderabbitai review

**Comment by ymahlau** [comment]
Thanks @bruxillensis for this great PR! I don't have much to say, most of the code already looks really good to me. Only a few minor comments from me and the AI review are listed above :)

File: src/fdtdx/objects/detectors/mode.py (hunk 2 above)
Comment (bruxillensis) [reply src/fdtdx/objects/detectors/mode.py:59]: This is done with the function "apply_mode_symmetry" in symmetry.py and is done automatically when symmetry is used in fdtdx simulation config. mode.py has the explicit parameter that can be passed to tidy3d when needed, but its handled automatically otherwise.

File: src/fdtdx/fdtd/symmetry.py [hunk 3]
```diff
@@ -0,0 +1,703 @@
...
+        sign = 1.0 if signs is None else signs.get(a, 1.0)
+        mirror = jnp.flip(arr, axis=ax) * sign
+        arr = jnp.concatenate([mirror, arr], axis=ax)
+    return arr
+
+
+def _stored_component_spec(components) -> list[tuple[Literal["E", "H"], int]]:
+    """Stored ``(field_type, component_axis)`` order for a Field/Phasor detector's components."""
+    return [_COMPONENT_SPEC[i] for i, name in enumerate(_COMPONENT_NAMES) if name in components]
+
```
[the remark this answers is not in the document]
Comment (bruxillensis) [reply src/fdtdx/fdtd/symmetry.py:460]: Both Field and Phasor detectors store components in canonical order (field.py:44-54 / phasor.py:88-99 test if "Ex" in components in fixed Ex,Ey,Ez,Hx,Hy,Hz sequence), regardless of the user's ordering. The current canonical-order spec correctly matches the stored array.

File: tests/integration/fdtd/test_symmetry_reduction.py [hunk 4]
```diff
@@ -0,0 +1,229 @@
...
+        object_list=objects, config=config, constraints=constraints, key=key
+    )
+
+    # Volume halved on y and z, unchanged on x.
+    assert oc.volume.grid_shape == (20, 10, 10)
+    assert arrays.fields.E.shape == (3, 20, 10, 10)
+
+    # PEC wall on y, PMC wall on z, each a single cell on the symmetry plane (min edge).
+    pec = oc.pec_objects
+    pmc = oc.pmc_objects
```
[the remark this answers is not in the document]
Comment (bruxillensis) [reply tests/integration/fdtd/test_symmetry_reduction.py:110]: These are @property (container.py:86-91). Current attribute access is correct; calling them as methods would try to invoke a list.

**Comment by bruxillensis** [comment]
I think this is ready to go! The test routine failed but I think thats a api key issue or something.

File: src/fdtdx/fdtd/symmetry.py (hunk 1 above)
Comment (bruxillensis) [reply src/fdtdx/fdtd/symmetry.py]: Added to core/misc.py

**Comment by ymahlau** [comment]
Looks good to me! Thanks @bruxillensis
